config/postcreate-hooks/kubestellar.yaml

```diff
@@ -189,6 +189,23 @@ spec:
               - "ControlPlaneName={{.ControlPlaneName}}"
             env:
             - name: XDG_CACHE_HOME
-              value: /tmp/helm/.cache  
+              value: /tmp/helm/.cache
+          - name: "{{.HookName}}-otp"
+            image: quay.io/kubestellar/helm:3.14.0
+            imagePullPolicy: IfNotPresent
+            args:
+              - upgrade
+              - --install
+              - --namespace
+              - "{{.Namespace}}"
+              - ocm-transport-plugin
+              - oci://ghcr.io/kubestellar/ocm-transport-plugin/chart/ocm-transport-plugin
+              - --version
+              - "0.1.7"
+              - --set
+              - "wds_cp_name={{.ControlPlaneName}}"
```

Comment:
Shouldn't this also `--set transport_cp_name=` something?

Reply:
@MikeSpreitzer Both ks and otp plugin can use an ITS parameter. In the current ks PCH deploying only the kc plugin, it has been decided not to pass explicitly an its parameter and let the plugin choose the its cp. I have adopted the same behavior when adding the otp plugin. You may remember that  a [PR](https://github.com/kubestellar/ocm-transport-plugin/pull/12) for OTP was used to allow for the automatic selection of the ITS if none is passed.

Reply:
> Both ks and otp plugin can use an ITS parameter

I suppose that by "can" you mean "could be changed to".

I am aware of the other facts that you recited. The problem is that we are trying to support a general number of ITSes and general choice of ITS for each WDS. Forcing a guess for every WDS does not do that. This is why I am advocating adding an ITS name parameter to the kubestellar PCH.

Reply:
@MikeSpreitzer there is no technical issue preventing it. the kflex command line will get longer and more verbose, even in the simple cases since there are no defaults, e.g.:

```shell
kflex create wds1 -p kubestellar --set ITSName=its1
```
There is also no easy way to document the set values since there "hidden" in the PCH yaml... perhaps we can put some comments at the header of the PCH.

I think in the short term we will move to use ks hub chart for the common setup and wih that it will easy, in my opinion, to achieve this and document it. Perhaps we can leave the current approach for a little longer and wait for the hub.
You can take a look to some initial drafting already enabling it [here](https://github.com/francostellari/kubestellar/tree/hub-chart/hub-chart)... please ignore the patches and test code to work around to pending new chart releases, this is not the final product

Reply:
I think that we can pursue the ITS-specification issue in follow-on PRs.

Reply:
@pdettori, I think that @francostellari has identified some important issues for PostCreateHooks:

1. Default values
2. Easy-to-find declaration of all the user-defined parameters

Reply:
I see your point @MikeSpreitzer. I am aware of the issue for default values  https://github.com/kubestellar/kubeflex/issues/233, I think it's a useful feature to have, we could also have some place for declarations. For the last point on helm, yes, that is a good point and I suppose it's true in general, so it seems like helm as a front-end for handling values and defaults makes a lot of sense.